plotly/tools.py

```diff
@@ -191,30 +198,46 @@ def set_config_file(plotly_domain=None,
     settings = get_config_file()
     if isinstance(plotly_domain, six.string_types):
         settings['plotly_domain'] = plotly_domain
+    elif plotly_domain is not None:
+        raise TypeError('Input should be a string')
```

Comment:
this seems like a backwards incompatible change. i think it's better this way.. i like it :)